pyo3-macros-backend/src/pymethod.rs

```diff
@@ -586,6 +586,34 @@ pub fn impl_py_setter_def(
         }
     };
 
+    let extract = if let PropertyType::Function { spec, .. } = &property_type {
+        Some(spec)
+    } else {
+        None
+    }
+    .and_then(|spec| {
+        let (_, args) = split_off_python_arg(&spec.signature.arguments);
+        let value_arg = &args[0];
+        let from_py_with = &value_arg.attrs.from_py_with.as_ref()?.value;
+        let name = value_arg.name.to_string();
+
+        Some(quote_spanned! { from_py_with.span() =>
+            let e = #pyo3_path::impl_::deprecations::GilRefs::new();
+            let from_py_with = #pyo3_path::impl_::deprecations::inspect_fn(#from_py_with, &e);
+            e.from_py_with_arg();
+            let _val = #pyo3_path::impl_::extract_argument::from_py_with(
+                &_value.into(),
+                #name,
+                from_py_with as fn(_) -> _,
+            )?;
+        })
+    })
+    .unwrap_or_else(|| {
+        quote! {
+            let _val = #pyo3_path::FromPyObject::extract_bound(_value.into())?;
```

Comment:
I suppose here we probably want to check if `_val` is a GIL Ref, via `holders.push_gil_refs_checker`?

Reply:
(Either way, I think worthy of its own PR and newsfragment, so I'll merge this PR now.)

Reply:
True, maybe I'll take a look at that tomorrow